apps/communications/contacts/test/unit/views/form/form_ui_test.js

```diff
@@ -0,0 +1,521 @@
+requireApp('communications/contacts/js/tag_options.js');
+require('/shared/js/text_normalizer.js');
+require('/shared/js/contacts/import/utilities/status.js');
+require('/shared/js/contacts/utilities/dom.js');
+require('/shared/js/contacts/import/utilities/misc.js');
+
+requireApp('communications/contacts/views/form/js/form_ui.js');
+
+/*
+ * NOTE: How tests are working:
+ * Every suite will try to test all actions, step by step, that
+ * you can execute in the form.
+ *
+ * Take into account that all tests within a suite rely on the former
+ * one, so we are keeping the state (in order to execute several UI
+ * actions in a linear manner).
+ *
+ * Keep in mind this behaviour!
+ *
+ */
+
+
+
+suite('FormUI', function() {
+
+  var realMozL10n;
+
+  suiteSetup(function(done) {
+
+    realMozL10n = navigator.mozL10n;
+    navigator.mozL10n = MockL10n;
+
+    // Load HTML
+    loadBodyHTML('/contacts/views/form/form.html');
+
+    // Add hook to template to "head"
+    var importHook = document.createElement('link');
+    importHook.id = 'form-import-link';
+    importHook.setAttribute('rel', 'import');
+    importHook.setAttribute('href', '/contacts/elements/form.html');
+    document.head.appendChild(importHook);
+
+    // Fill the HTML
+    LazyLoader.load([document.getElementById('view-contact-form')], function() {
+      FormUI.init();
+      done();
+    });
+  });
+
+  suiteTeardown(function() {
+    navigator.mozL10n = realMozL10n;
+    realMozL10n = null;
+    var importHook =
+      document.querySelector('link[rel=import][id="form-import-link"]');
+    document.head.removeChild(importHook);
+    document.body.innerHTML = '';
+    window.FormUI = null;
+  });
+
+  suite('Close button', function() {
+    test(' > must dispatch an event when clicked', function(done) {
+      window.addEventListener('close-ui', function() {
+        done();
+      });
+
+      var clickEvent = new CustomEvent('action');
+      document.querySelector('#contact-form-header').dispatchEvent(clickEvent);
+    });
+  });
+
+
+  suite('Phone number', function() {
+    var contact = {
+      tel: ['123456789']
+    };
+
+    test(' > Add one phone number by "render" method', function() {
+      FormUI.render(contact);
+      var inputsRendered =
+        document.querySelectorAll(
+          'input[data-field="value"]:not([value=""]):not([value="#value#"])'
+        );
+      inputsRendered = Array.prototype.filter.call(
+        inputsRendered,
+        function(element) {
+          return element.id.startsWith('number');
+        }
+      );
+
+      assert.equal(inputsRendered.length, 1);
+      assert.equal(inputsRendered[0].value, contact.tel[0]);
+    });
+
+    test(' > Save button must be enabled', function() {
+      var saveButton = document.querySelector('#save-button');
+      assert.equal(saveButton.getAttribute('disabled'), null);
+    });
+
+    test(' > Reset in given phone number', function() {
+      var inputs =
+        document.querySelectorAll(
```

Comment:
Should we have an utility function to wrap this querySelectorAll? It will add some readability:
```
var inputs = getInputs();
```